### PR TITLE
Add mapping tests

### PR DESCRIPTION
**Pull request by guibescos**
add_mapping test ported to rust.

The pubkey operations are really horrible right now. They got implemented in pyth-solana-sdk, so I feel like we'll be able to fix them once we move from bindgen to pyth-solana-sdk.

### CODE REVIEW

File: program/rust/src/tests/test_add_mapping.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        ],
+        instruction_data
+    )
+    .is_ok());
+
+    {
+        let next_mapping_data = load_mapping_account_mut(&next_mapping, PC_VERSION).unwrap();
+        let mut cur_mapping_data = load_mapping_account_mut(&cur_mapping, PC_VERSION).unwrap();
+
+        assert!(unsafe {
```
Comment (guibescos) [inline program/rust/src/tests/test_add_mapping.rs:105]: The pubkey operations are really horrible right now. They got implemented in pyth-solana-sdk, so I feel like we'll be able to fix them once we move from bindgen to pyth-solana-sdk.

Comment (guibescos) [reply program/rust/src/tests/test_add_mapping.rs:105]: I changed my mind, I'll implement them and when we move to pyth-solana-sdk we can replace the functions.

Comment (jayantk) [reply program/rust/src/tests/test_add_mapping.rs:105]: Ok. I think you can merge this PR now and update the tests with those functions in a separate pr.

**Review by jayantk — APPROVED** [review]
looks great.

Something to think about going forward: can we implement some helpers for writing tests more concisely? The tests are rather verbose right now, and that makes it hard for us to test everything thoroughly.

**Pull request merged**